barretenberg/cpp/src/barretenberg/ultra_honk/decider_proving_key.hpp

```diff
@@ -200,10 +200,14 @@ template <IsUltraOrMegaHonk Flavor> class DeciderProvingKey_ {
                 public_inputs.emplace_back(polynomials.w_r[idx]);
             }
 
-            // Set the pairing point accumulator indices. This should exist for all flavors.
-            ASSERT(circuit.pairing_inputs_public_input_key.is_set() &&
-                   "Honk circuit must output a pairing point accumulator. If this is a test, you might need to add a \
+            // Set the pairing point accumulator indices. This should exist for all non-Mega Flavors.
+            if constexpr (IsUltraHonk<Flavor>) {
+                ASSERT(
+                    circuit.pairing_inputs_public_input_key.is_set() &&
+                    "Honk circuit must output a pairing point accumulator. If this is a test, you might need to add a \
                    default one through a method in PairingPoints.");
+            }
+            // TODO(https://github.com/AztecProtocol/barretenberg/issues/1468): Remove use of pairing inputs pub key
             metadata.pairing_inputs_public_input_key = circuit.pairing_inputs_public_input_key;
```

Comment:
Somewhat confusingly, we still need to set this value for Mega _sometimes_. I elaborate a bit in the linked issue. Planning to follow up on this soon